src/main.ts

```diff
@@ -1,9 +1,37 @@
 import { Plugin, MarkdownView, Editor } from "obsidian";
+import {CheckboxReorderPluginSettingTab} from './settings'
 import { reorderCheckboxesInFile } from "src/reorderCheckboxes";
 
+interface CheckboxReorderPluginSettings {
+	autoReorder: boolean;
+}
+
+const DEFAULT_SETTINGS: Partial<CheckboxReorderPluginSettings> = {
+	autoReorder: false
+}
+
 export default class CheckboxReorderPlugin extends Plugin {
+	settings: CheckboxReorderPluginSettings;
+
 	async onload() {
+		await this.loadSettings();
+
 		console.log("Checkbox Reorder Plugin loaded");
+		
+
+		this.addSettingTab(new CheckboxReorderPluginSettingTab(this.app, this))
+
+		this.registerEvent(this.app.vault.on("modify", () => {
+			
+			if(this.settings.autoReorder){
+				const view = this.app.workspace.getActiveViewOfType(MarkdownView);
+				const cursorPosition = view?.editor.getCursor()
+				if(view?.editor && cursorPosition){
+					this.reorderCheckboxes(view?.editor);
+					view?.editor.setCursor(cursorPosition);
```

Comment:
what happens when you don't set the cursor position?

Reply:
If you don't set the cursor position, ever time `reorderCheckboxes()` is called, the cursor is set to the start of the file. 

Essentially every time you type something, or modify the file in any way, the cursor would be set to the start of the file.

Reply:
I see. Thank you for the explanation!